### PR TITLE
Add base views

### PR DESCRIPTION
**Pull request by baru64**

### CODE REVIEW

File: jolly-jellyfish/src/django_meta/settings.py [hunk 1]
```diff
@@ -8,8 +8,10 @@
 # Build paths inside the project like this: os.path.join(BASE_DIR, ...)
 BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
 MEDIA_ROOT = os.path.join(BASE_DIR, 'user_media')
+MEDIA_URL = "/user_media/"
```
Comment (A5rocks) [inline jolly-jellyfish/src/django_meta/settings.py:11]: We might want to use single quotes...

File: jolly-jellyfish/src/page_maker/templates/page_maker/template_detail.html [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{% extends "page_maker/base.html" %}
+
+{% block header %}
+<link rel="stylesheet" type="text/css" href="{{ template.style_sheet.url }}">
+{% endblock %}
+
+{% block content %}
+<div><h1>{{ template.name }}</h1></div>
+<!-- <img src=""> -->
```
Comment (A5rocks) [inline jolly-jellyfish/src/page_maker/templates/page_maker/template_detail.html:9]: What's with these commented out images?

File: jolly-jellyfish/src/page_maker/templates/page_maker/template_detail.html [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+<div>user text 1</div>
+<!-- <img src=""> -->
```
Comment (A5rocks) [inline jolly-jellyfish/src/page_maker/templates/page_maker/template_detail.html:11]: What's with these commented out images?

File: jolly-jellyfish/src/page_maker/templates/page_maker/template_detail.html [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+<div>user text 2</div>
+<!-- <img src=""> -->
```
Comment (A5rocks) [inline jolly-jellyfish/src/page_maker/templates/page_maker/template_detail.html:13]: What's with these commented out images?

File: jolly-jellyfish/src/page_maker/templates/page_maker/template_detail.html [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+<div>user text 3</div>
+<!-- <img src=""> -->
```
Comment (A5rocks) [inline jolly-jellyfish/src/page_maker/templates/page_maker/template_detail.html:15]: What's with these commented out images?

File: jolly-jellyfish/src/page_maker/templates/page_maker/user_detail.html [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+  {% endif %}
+</div>
+<div>
+  {% for webpage in user_pages %}
+  <div>
+    <a href="{% url 'webpage-view' pagename=webpage.name %}">
+      <p>{{ webpage.user_title }}</p>
+    </a>
+    {% if webpage.thumbnail %}
+    <img src="{{ webpage.thumbnail.url }}">
```
Comment (A5rocks) [inline jolly-jellyfish/src/page_maker/templates/page_maker/user_detail.html:18]: Should we indent for legibility?

File: jolly-jellyfish/src/page_maker/templates/page_maker/user_detail.html [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    {% else %}
+    no thumbnail
```
Comment (A5rocks) [inline jolly-jellyfish/src/page_maker/templates/page_maker/user_detail.html:20]: Same as above.

File: jolly-jellyfish/src/page_maker/templates/page_maker/webpage_delete.html [hunk 8]
```diff
@@ -0,0 +1,14 @@
+{% extends "page_maker/base.html" %}
+
+{% block content %}
+<div>
+  <form method="POST">
+  {% csrf_token %}
+    <div>
+      {{ form.as_p }}
+      <p>Delete page <b>{{webpage.name}}</b></p>
```
Comment (A5rocks) [inline jolly-jellyfish/src/page_maker/templates/page_maker/webpage_delete.html:9]: Should this maybe have a question mark? Like:

Delete page **xyzzy**?

File: jolly-jellyfish/src/page_maker/templates/page_maker/webpage_detail.html [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+  {% endfor %}
+</div>
+<div>
+  <form method="POST" enctype="multipart/form-data" action="{% url 'comment-create' pagename=webpage.name %}">
+    {% csrf_token %}
+      <div>
+        {{ comment_form.as_p }}
+        <button type="submit">Add comment</button>
+      </div>
+    </form>
```
Comment (A5rocks) [inline jolly-jellyfish/src/page_maker/templates/page_maker/webpage_detail.html:28]: 1 too many indents

File: jolly-jellyfish/src/page_maker/views.py [hunk 10]
```diff
@@ -18,3 +29,214 @@ class UserRegister(FormView):
...
+
+
+class UserDetailView(DetailView):
+    model = User
+    context_object_name = 'viewed_user'
+    template_name = 'page_maker/user_detail.html'
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        # user = get_object_or_404(User, pk=self.kwargs.get('pk'))
```
Comment (A5rocks) [inline jolly-jellyfish/src/page_maker/views.py:41]: Perhaps an extra comment?

File: jolly-jellyfish/src/page_maker/templates/page_maker/webpage_list.html [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+  <div>
+    <a href="{% url 'webpage-view' pagename=webpage.name %}">
+      <p>{{ webpage.user_title }}</p>
+    </a>
+    {% if webpage.thumbnail %}
+    <img src="{{ webpage.thumbnail.url }}">
+    {% else %}
+    no thumbnail
+    {% endif %}
+  </div>
```
Comment (A5rocks) [inline jolly-jellyfish/src/page_maker/templates/page_maker/webpage_list.html:14]: Is this an extra indent?

File: jolly-jellyfish/src/page_maker/templates/page_maker/template_list.html [hunk 12]
```diff
@@ -0,0 +1,11 @@
+{% extends "page_maker/base.html" %}
+
+{% block content %}
+{% for template in templates %}
+  <div>
+    <a href="{% url 'template-detail' pk=template.id %}">
+      <p>{{ template.name }}</p>
+    </a>
+  </div>
```
Comment (A5rocks) [inline jolly-jellyfish/src/page_maker/templates/page_maker/template_list.html:9]: One too many indents?

**Review by A5rocks — COMMENTED** [review]
Welp I was too late... It's only stylistic comments anyway.

File: jolly-jellyfish/src/page_maker/templates/page_maker/template_detail.html (hunk 2 above)
Comment (baru64) [reply jolly-jellyfish/src/page_maker/templates/page_maker/template_detail.html:9]: I was supposed to put there some placeholder images for template preview but I forgot.

**Pull request merged**